src/services/futures/trade.ts

```diff
@@ -1,15 +1,25 @@
-import { ethers } from "../..";
-import { FUTURES_TRACKING, PRICE_IMPACT_DELTA } from "./constants";
+import { Pool, ethers } from "../..";
+import { FUTURES_TRACKING } from "./constants";
 import ISynthetixFuturesMarketV2 from "../../abi/ISynthetixFuturesMarketV2.json";
 
-export function getFuturesChangePositionTxData(
-  amount: ethers.BigNumber | string
-): string {
+export async function getFuturesChangePositionTxData(
+  amount: ethers.BigNumber | string,
+  market: string,
+  pool: Pool
+): Promise<string> {
+  const futuresMarket = new ethers.Contract(
+    market,
+    ISynthetixFuturesMarketV2.abi,
+    pool.signer
+  );
+  const fillPrice = await futuresMarket.fillPrice(amount);
+  const adjustmentFactor = ethers.BigNumber.from(amount).lt(0) ? 995 : 1005;
```

Comment:
Maybe a comment would be helpful eg
"Allows for +-0.5% price movements on the desired fill price"

Reply:
added comment